src/mongo/db/dbcommands_generic.cpp

```diff
@@ -275,12 +275,18 @@ class ListCommandsCmd : public Command {
                      int,
                      string& errmsg,
                      BSONObjBuilder& result) {
+        // sort the commands before building the result BSON
+        std::map<string, Command*> sortedCommands;
+        for (CommandMap::const_iterator it = _commands->begin(); it != _commands->end(); ++it) {
+            sortedCommands.emplace(it->first, it->second);
+        }
```

Comment:
Sorry, I mistyped that. It should be

```
std::map<CommandMap::key_type, CommandMap::mapped_type>
```
Re performance: this is somewhat academic, because the actual performance difference will be unobservable, and this is not exactly hot-path code.

However, vector is often faster than map in practice due to its low constant factors and better cache utilization, since it is a contiguous container rather than a node based container.